File: java/src/LC_283_MoveZeroes.java

import java.util.Arrays;

// https://leetcode.com/problems/move-zeroes/
public class LC_283_MoveZeroes {
    public void moveZeroes(int[] nums) {
        int index = 0;
        for (int num : nums) {
            if (num != 0)
                nums[index++] = num;
        }

        for (int i = index; i < nums.length; i++) {
            nums[i] = 0;
        }
    }

    public static void main(String[] args) {
        LC_283_MoveZeroes moveZeroes = new LC_283_MoveZeroes();
        int[] nums = { 1, 2, 0, 4, 0, 0, 6, 7, 0 };
        moveZeroes.moveZeroes(nums);
        System.out.println(Arrays.toString(nums));
    }

}

File: python/two_pointers/LC_283_MoveZeroes.py
# https://leetcode.com/problems/move-zeroes/
from typing import List

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        index = 0
        for num in nums:
            if num != 0:
                nums[index] = num
                index += 1
        while index <len(nums):
            nums[index] = 0
            index += 1

sol = Solution()
nums  = [1,2, 0, 4, 0, 0, 6, 0, 9]
sol.moveZeroes(nums)
print(nums)